### PR TITLE
feat(Jax-frontend): Add histogramdd to JAX Frontend 

### LINKED ISSUE #26298: histogramdd

### PR DESCRIPTION
**Pull request by ThanhDuonghcmut**
# PR Description

## Related Issue #26298

Close #26298

## Checklist

- [x] Did you add a function?
- [x] Did you add the tests?
- [x] Did you run your tests and are your tests passing?
- [x] Did pre-commit not fail on any check?
- [x] Did you follow the steps we provided?

### CODE REVIEW

**Comment by aparajith21** [comment]
Hey, thanks for contributing! Please let me know once you've ticked off the items in the checklist and this is ready for review!

**Comment by ThanhDuonghcmut** [comment]
Could you kindly help me to understand what this warning means? Does it affect my code quality?

 
And also, as my code runs with JAX frontend histogramdd, could I ask why I got this error about numpy_exp2?

I really appreciate your reviews. Please kindly help me clarify them. Thank you very much.

**Comment by ThanhDuonghcmut** [comment]
I've checked all the ticks on the items. Could you please give me a review for my code?

**Comment by ThanhDuonghcmut** [comment]
Hi @aparajith21 , could you kindly have a review for my code?

**Comment by aparajith21** [comment]
Thanks for contributing! All seems good, test failures seem unrelated to PR, but could you please remove usage of external libraries like `math` and `operator` and substitute their usage with either built in python functions or ivy functions

File: ivy/functional/frontends/jax/numpy/statistical.py [hunk 1]
```diff
@@ -9,6 +9,11 @@
 from ivy.functional.frontends.jax.numpy import promote_types_of_jax_inputs
 
 
+import builtins
+import operator
+import math
```
Comment (aparajith21) [inline ivy/functional/frontends/jax/numpy/statistical.py:14]: Could the usage of these libraries be replaced with ivy functions please

**Comment by ThanhDuonghcmut** [comment]
Hi @aparajith21 , I've replaced the external libraries with built-in Python or Ivy functions. However, it produced more failures. Could you kindly have a re-review of what is happening and help me to explain the reason of my failures, as I don't understand what the failures said? 
Thank you very much.

**Comment by ThanhDuonghcmut** [comment]
Hello @aparajith21 , could you please review my code? I've removed the external libs as you requested

**Comment by ThanhDuonghcmut** [comment]
Hello @Fayad-Alman, could you please review my code? Thank you very much

**Comment by Ishticode** [comment]
waiting on fresh ci run

**Comment by ivy-seed** [comment]
This PR has been labelled as stale because it has been inactive for more than 7 days. If you would like to continue working on this PR, then please add another comment or this PR will be closed in 7 days.

**Review by Ishticode — CHANGES_REQUESTED** [review]
Hi @ThanhDuonghcmut 
Thanks for the PR.
It seems like there is failure caught by the CI has caught a failure for test_jax_histogramd as you can check here at https://github.com/unifyai/ivy/actions/runs/8129963424/job/22217724147?pr=26725. Please try to fix and request a review when done. Thank you :)

**Comment by Ishticode** [comment]
Closing this due to inactivity. Please feel free to reopen if you would like to continue working on this. Thanks :)
